File: src/main/kotlin/de/thm/mni/ii/classroom/event/InvitationEvent.kt
package de.thm.mni.ii.classroom.event

import com.fasterxml.jackson.annotation.JsonIgnore
import de.thm.mni.ii.classroom.model.classroom.ConferenceInfo
import de.thm.mni.ii.classroom.model.classroom.UserCredentials

data class InvitationEvent(
    val inviter: UserCredentials,
    val invitee: UserCredentials,
    val conferenceInfo: ConferenceInfo
) : ClassroomEvent(InvitationEvent::class.simpleName!!) {
    @JsonIgnore
    override fun getClassroomId(): String {
        return if (inviter.classroomId !== invitee.classroomId || inviter.classroomId !== conferenceInfo.classroomId) {
            "INVALID"
        } else {
            this.conferenceInfo.classroomId
        }
    }
}

File: web-gui/src/app/rsocket/event/InvitationEvent.ts
import {UserCredentials} from "../../model/User";
import {ConferenceInfo} from "../../model/ConferenceInfo";
import {ClassroomEvent} from "./ClassroomEvent";

export class InvitationEvent extends ClassroomEvent {
    inviter: UserCredentials | undefined
    invitee: UserCredentials | undefined
    conferenceInfo: ConferenceInfo | undefined

    constructor() {
        super("InvitationEvent");
    }
}
